src/clinical/clinical-service.ts

```diff
@@ -240,19 +243,84 @@ export const getClinicalEntityMigrationErrors = async (programId: string, query:
           donor.programId.toString() === programId && query.includes(donor.donorId.toString()),
       )
       .forEach(donor => {
-        const { donorId, submitterDonorId } = donor;
+        const { donorId, submitterDonorId, errors } = donor;
         // Overwrite donor.errors + flatten entityName to simplify query
-        let errors: DonorMigrationErrorRecord[] = [];
-        donor.errors.forEach(entity => {
-          const entityName = Object.keys(entity)[0];
-          errors = errors.concat(entity[entityName].map(error => ({ ...error, entityName })));
+        // Input: Donor.Errors = [{ [entityName] : [{error}] }]
+        // =>  Output: Donor.Errors = [{ ...error, entityName}]
+
+        errors.forEach(entityErrorObject => {
+          const currentEntityErrorData: [
+            string | EntityAlias,
+            readonly DeepReadonly<SchemaValidationError>[],
+          ] = Object.entries(entityErrorObject)[0];
+
+          const entityName = currentEntityErrorData[0] as EntityAlias;
+          const entityErrors = currentEntityErrorData[1] as readonly DeepReadonly<
+            SchemaValidationError
+          >[];
+
+          const updatedErrorEntries = entityErrors.map(error => ({
+            ...error,
+            donorId,
+            entityName,
+          }));
+
+          const updatedDonorErrorData: ClinicalErrorsResponseRecord = {
+            donorId,
+            submitterDonorId,
+            errors: updatedErrorEntries,
+          };
+
+          clinicalMigrationErrors.push(updatedDonorErrorData);
         });
-        clinicalErrors.push({ donorId, submitterDonorId, errors });
       });
   }
-
   const end = new Date().getTime() / 1000;
   L.debug(`getClinicalEntityMigrationErrors took ${end - start}s`);
 
-  return clinicalErrors;
+  return { clinicalErrors: clinicalMigrationErrors, migrationLastUpdated };
+};
+
+export const getDonorSubmissionErrorUpdates = async (
```

Comment:
This function (and also the `getClinicalEntityMigrationErrors`) could benefit from some comments (JSDocs preferable) in order to explain what data these are returning.

Reply:
OK I've updated comments + added return types to functions